pkg/controllers/report/utils/scanner.go

```diff
@@ -44,30 +46,39 @@ func NewScanner(
 	}
 }
 
-func (s *scanner) ScanResource(ctx context.Context, resource unstructured.Unstructured, nsLabels map[string]string, policies ...kyvernov1.PolicyInterface) map[kyvernov1.PolicyInterface]ScanResult {
-	results := map[kyvernov1.PolicyInterface]ScanResult{}
-	for _, policy := range policies {
+func (s *scanner) ScanResource(ctx context.Context, resource unstructured.Unstructured, nsLabels map[string]string, policies ...engineapi.GenericPolicy) map[*engineapi.GenericPolicy]ScanResult {
+	results := map[*engineapi.GenericPolicy]ScanResult{}
+	for i, policy := range policies {
 		var errors []error
 		logger := s.logger.WithValues("kind", resource.GetKind(), "namespace", resource.GetNamespace(), "name", resource.GetName())
-		response, err := s.validateResource(ctx, resource, nsLabels, policy)
-		if err != nil {
-			logger.Error(err, "failed to scan resource")
-			errors = append(errors, err)
-		}
-		spec := policy.GetSpec()
-		if spec.HasVerifyImages() {
-			ivResponse, err := s.validateImages(ctx, resource, nsLabels, policy)
+		var response *engineapi.EngineResponse
+		if policy.GetType() == engineapi.KyvernoPolicyType {
+			var err error
+			pol := policy.GetPolicy().(kyvernov1.PolicyInterface)
+			response, err = s.validateResource(ctx, resource, nsLabels, pol)
 			if err != nil {
-				logger.Error(err, "failed to scan images")
+				logger.Error(err, "failed to scan resource")
 				errors = append(errors, err)
```

Comment:
Should we continue here?

Reply:
Do we want to continue validating images if fail to validate resources?
```
ivResponse, err := s.validateImages(ctx, resource, nsLabels, pol)
```

Reply:
No because if an error is found, we abort it when looping through the `ScanResult`. I will abort it from the Scanner instead!